### PR TITLE
[math] Use Math Transformation Services as Profiles

### PR DESCRIPTION
**Pull request by cweitkamp**
- Use Math Transformation Services as Profiles

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.smarthomej.transform.math/README.md [hunk 1]
```diff
@@ -7,9 +7,16 @@ Transforms the input by applying simple math on it.
...
-Number secondsToMinutes "Time [DIVIDE(60):%s]" { channel="xxx" }
+Number multiply "Value multiplied by [MULTIPLY(1000):%s]" { channel="<channelUID>" }
+Number add "Value added [ADD(5.1):%s]" { channel="<channelUID>" }
+Number secondsToMinutes "Time [DIVIDE(60):%s]" { channel="<channelUID>" }
+Number subtracted "Value subtracted [ADD(-1):%s]" { channel="<channelUID>" }
+
+// Usage as a Profile
+Number multiply "Value multiplied by [%.1f]" { channel="<channelUID>" [profile="transform:MULTIPLY", multiplicand=1000] }
+Number add "Value added [%.1f]" { channel="<channelUID>" [profile="transform:ADD", addend=5.1] }
+Number secondsToMinutes "Time [%d]" { channel="<channelUID>" [profile="transform:DIVIDE, dvisor=60] }
```
Comment (J-N-K) [inline bundles/org.smarthomej.transform.math/README.md]:
```suggestion
Number secondsToMinutes "Time [%d]" { channel="<channelUID>" [profile="transform:DIVIDE, divisor=60] }
```

File: bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/binding/math/internal/AbstractMathTransformationService.java [hunk 2]
```diff
@@ -30,7 +30,8 @@
  */
 @NonNullByDefault
 abstract class AbstractMathTransformationService implements TransformationService {
-    private final Logger logger = LoggerFactory.getLogger(AbstractMathTransformationService.class);
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
```
Comment (J-N-K) [inline bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/binding/math/internal/AbstractMathTransformationService.java:34]: nice

File: bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/binding/math/internal/profiles/MathTransformationProfileFactory.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    private static final Stream<ProfileType> SUPPORTED_PROFILE_TYPES = Stream.of(ADD_PROFILE_TYPE,
+            MULTIPLY_PROFILE_TYPE, DIVIDE_PROFILE_TYPE);
+
+    private final Map<LocalizedKey, ProfileType> localizedProfileTypeCache = new ConcurrentHashMap<>();
+
+    private final ProfileTypeI18nLocalizationService profileTypeI18nLocalizationService;
+    private final Bundle bundle;
+
+    @NonNullByDefault({})
+    private TransformationService addTransformationService;
```
Comment (J-N-K) [inline bundles/org.smarthomej.transform.math/src/main/java/org/smarthomej/binding/math/internal/profiles/MathTransformationProfileFactory.java]:
```suggestion
    private @NonNullByDefault({}) TransformationService addTransformationService;
```

seems more consistent with the other null-annotations code style

File: bundles/org.smarthomej.transform.math/src/main/resources/OH-INF/i18n/transform_de.properties [hunk 4]
```diff
@@ -0,0 +1,11 @@
+profile-type.transform.ADD.label = Addieren
+profile.config.transform.ADD.addend.label = Summand
+profile.config.transform.ADD.addend.description = Wert, der zu dem item state addiert wird.
```
Comment (J-N-K) [inline bundles/org.smarthomej.transform.math/src/main/resources/OH-INF/i18n/transform_de.properties]:
```suggestion
profile.config.transform.ADD.addend.description = Wert, der zu dem Item State addiert wird.
```

**Review by J-N-K — CHANGES_REQUESTED** [review]
Nice, I have left some comments.

**Comment by cweitkamp** [comment]
Thanks for your quick review. I added unit tests, renamed packages and incorporated your comments from review .

**Review by J-N-K — APPROVED** [review]

**Pull request merged**